using builder pattern and ubuntu-18.04 upgrade

enricomarchesin opened the pull request:
upgraded ubuntu image to 18.04 to have a slimmer image (https://blog.ubuntu.com/2018/07/09/minimal-ubuntu-released) and switched to builder pattern for sources compilation (https://blog.alexellis.io/mutli-stage-docker-builds/)

Ullaakut commented on .travis.yml:
```diff
@@ -9,7 +9,7 @@ before_install:
   - echo "Docker Hub credentials are working"
 
 script:
-  - ./build.sh
```
Could you also remove the `build.sh` script since it's no longer useful?

enricomarchesin commented:
Addressed the comment above.

Indeed the "ubuntu:18.04" is **very** optimised: I also tried with the, supposedly "super slim" `minideb` base image from Bitnami, but with the official one we can shave another 100 MB from the final image, down to close to 500 MB.

I'm sure you know the original image based on ubuntu:17.10 was 1000 MB in size: a 50% reduction modifying just a couple of lines! ☺️

Ullaakut commented:
> Addressed the comment above.
> 
> Indeed the "ubuntu:18.04" is **very** optimised: I also tried with the, supposedly "super slim" `minideb` base image from Bitnami, but with the official one we can shave another 100 MB from the final image, down to close to 500 MB.
> 
> I'm sure you know the original image based on ubuntu:17.10 was 1000 MB in size: a 50% reduction modifying just a couple of lines! ☺️

Looks good to me! Thanks for your contribution, once again! 🎉
